FinniversKit/Sources/SwiftUI Components/Button/SwiftUIIconButton.swift

```diff
@@ -0,0 +1,40 @@
+import SwiftUI
+
+public struct SwiftUIIconButton: View {
+    public struct Style {
+        let icon: ImageAsset
+        let iconToggled: ImageAsset
+    }
+
+    @Binding public var isToggled: Bool
+    private let style: Style
+
+    public init(style: Style, isToggled: Binding<Bool>) {
+        self.style = style
+        self._isToggled = isToggled
+    }
+
+    public var body: some View {
+        Image.init(isToggled ? style.iconToggled : style.icon)
+            .accessibility(addTraits: isToggled ? [.isSelected] : [])
```

Comment:
A bit late to the party, since this is merged, but I'll leave this either way😄

- [`accessibility(addtraits:)` is deprecated](https://developer.apple.com/documentation/swiftui/view/accessibility(addtraits:)#).
    - We should use [`accessibilityAddTraits(_:)`](https://developer.apple.com/documentation/swiftui/view/accessibilityaddtraits(_:)) instead.
- I think we should add `.isButton` to the list of traits as well, since all we have here is an `Image`.
    - `accessibilityAddTraits(isToggled ? [.isSelected, .isButton] : [.isButton])`